src/Features/CSharp/Portable/CodeFixes/GenerateVariable/GenerateVariableCodeFixProvider.cs

```diff
@@ -24,11 +24,10 @@ internal class GenerateVariableCodeFixProvider : AbstractGenerateMemberCodeFixPr
         private const string CS0117 = nameof(CS0117); // error CS0117: 'TestNs.Program' does not contain a definition for 'blah'
         private const string CS0539 = nameof(CS0539); // error CS0539: 'Class.SomeProp' in explicit interface declaration is not a member of interface
         private const string CS0246 = nameof(CS0246); // error CS0246: The type or namespace name 'Version' could not be found
+        private const string CS0120 = nameof(CS0120); // error CS0120: An object reference is required for the non-static field, method, or property 'A'
```

Comment:
THe design is unfortunate.  I don't mind tying us to the compiler's pass over the code to avoid having two systems churning over the world.  THe problem stems from us using diagnostics as the currency to try to back infer what we should do.  We really need to move to a mechanism that is more in line with our actual Compilation APIs (like GetSymbolInfo/GetTypeInfo).  For example, we could ask for a callback whenever GetSymbolInfo would return a specific candidate reason.  Right now we have an opaque "id + span" pair, and we work entirely off of that.  If the span isn't what we expect, things don't work.  If we dont' know about an ID, things don't work.  If the compiler suppresses a diagnostic, things don't work.  It's effectively moving us away from having a well designed and documented compiler API to use, to the moral equivalent of screen scraping.  It's not much better than us just runnign the compiler, seeing the emitted errors, and trying to go back from that to decide what to do.

I've discussed this issue with @srivatsn and @JohnHamby.  I really think the IOperation API should have a mechanism by which consumers can register actions to run *not* just on nodes, symbols, operations and tree, but *also* on strongly typed SymbolInfo information like 'CandidateReason'.  

In other words, given any node 'n', if calling semanticModel.GetSymbolInfo(n) would return a SymbolInfo 'si' with CandidateReason 'r', then i should be able to do something like "context.RegisterCandidateReason(r, callback)" and be called back with all relevant 'n's and 'si's.

This would allow us to move the IDE code back to what we had many years ago, with really clear and crisp code that used the actual compilation APIs, while still getting the perf benefit of running in tandem with the normal analysis pass.